x-pack/plugins/aiops/public/components/explain_log_rate_spikes/field_filter_popover.tsx

```diff
@@ -0,0 +1,221 @@
+  disabledApplyButton,
+  uniqueFieldNames,
+  onChange,
+}) => {
+  const euiThemeContext = useEuiTheme();
+  // Inspired by https://github.com/elastic/eui/blob/main/src/components/datagrid/controls/_data_grid_column_selector.scss
+  const fieldSelectPopover = useMemo(
+    () => css`
+      ${euiYScrollWithShadows(euiThemeContext, {})}
+      max-height: 400px;
+    `,
+    [euiThemeContext]
+  );
+
+  const [fieldSearchText, setFieldSearchText] = useState('');
+  const [skippedFields, setSkippedFields] = useState<string[]>([]);
+  const setFieldsFilter = (fieldNames: string[], checked: boolean) => {
+    let updatedSkippedFields = [...skippedFields];
+    if (!checked) {
+      updatedSkippedFields.push(...fieldNames);
+    } else {
+      updatedSkippedFields = skippedFields.filter((d) => !fieldNames.includes(d));
+    }
+    setSkippedFields(updatedSkippedFields);
+  };
+
+  const [isFieldSelectionPopoverOpen, setIsFieldSelectionPopoverOpen] = useState(false);
+  const onFieldSelectionButtonClick = () => setIsFieldSelectionPopoverOpen((isOpen) => !isOpen);
+  const closePopover = () => setIsFieldSelectionPopoverOpen(false);
+
+  const filteredUniqueFieldNames = useMemo(() => {
+    return uniqueFieldNames.filter(
+      (d) => d.toLowerCase().indexOf(fieldSearchText.toLowerCase()) !== -1
+    );
+  }, [fieldSearchText, uniqueFieldNames]);
+
+  // If the supplied list of unique field names changes, do a sanity check to only
+  // keep field names in the list of skipped fields that still are in the list of unique fields.
+  useEffect(() => {
+    setSkippedFields((previousSkippedFields) =>
+      previousSkippedFields.filter((d) => uniqueFieldNames.includes(d))
+    );
+  }, [uniqueFieldNames]);
+
+  const selectedFieldCount = uniqueFieldNames.length - skippedFields.length;
+
+  return (
+    <EuiPopover
+      data-test-subj="aiopsFieldFilterPopover"
+      anchorPosition="downLeft"
+      panelPaddingSize="s"
+      button={
+        <EuiButton
+          data-test-subj="aiopsFieldFilterButton"
+          onClick={onFieldSelectionButtonClick}
+          disabled={disabled}
+          size="s"
+          iconType="arrowDown"
+          iconSide="right"
+          iconSize="s"
+        >
+          <FormattedMessage
+            id="xpack.aiops.explainLogRateSpikesPage.fieldFilterButtonLabel"
+            defaultMessage="Filter fields"
+          />
+        </EuiButton>
+      }
+      isOpen={isFieldSelectionPopoverOpen}
+      closePopover={closePopover}
+    >
+      <EuiPopoverTitle>
+        <EuiText size="xs" color="subdued" style={{ maxWidth: '400px' }}>
+          <FormattedMessage
+            id="xpack.aiops.explainLogRateSpikesPage.fieldFilterHelpText"
+            defaultMessage="To remove non-relevant fields from groups, deselect them below and click the Apply button to rerun the grouping. The search bar can be used to filter the list of fields and is useful to select/deselect in bulk with the actions below the list."
```

Comment:
```suggestion
            defaultMessage="Deselect non-relevant fields to remove them from groups and click the Apply button to rerun the grouping.  Use the search bar to filter the list, then select/deselect multiple fields with the actions below."
```

Reply:
Updated in [16a3325](https://github.com/elastic/kibana/pull/153864/commits/16a3325443116b763ffc15b9081f9beb383f5b1e).